### PR TITLE
codeowners: add obs-india-prs

### PR DESCRIPTION
**Pull request by dhartunian**
Add obs india team to relevant pkg/util and pkg/cli code that they own.

Release note: None

### CODE REVIEW

**Review by xinhaoz — APPROVED** [review]

**Comment by celiala** [comment]
Seeing this [error](https://mesolite.cluster.engflow.com/invocations/default/4bb175f4-5600-47c6-b54d-4a9e59ff3e31):
```
=== RUN   TestLint/CODEOWNERS
    gen-lint_test.go:2675: 
        	Error Trace:	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/testutils/lint/gen-lint_test.go:2675
        	Error:      	Received unexpected error:
        	            	owner cockroachdb/obs-india-prs does not exist
```
This error was [getting me for a while](https://github.com/cockroachdb/cockroach/pull/123920#issuecomment-2156693168): I believe [you'll want to map this to an Owner in owners.go](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/registry/owners.go#L26)

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -167,7 +167,7 @@
 /pkg/cli/tsdump.go           @cockroachdb/obs-prs
```
Comment (arjunmahishi) [inline .github/CODEOWNERS:167]: This one too?

**Review by arjunmahishi — APPROVED** [review]

**Review by aa-joshi — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127072#-:-O1ordh52i4hZBR04tjl:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @arjunmahishi and @dhartunian)

**Pull request merged**